src/libunicode/intrinsics.h

```diff
@@ -72,6 +73,144 @@ struct platform_intrinsics<__m128i>
 
 using intrinsics = platform_intrinsics<__m128i>;
 
+template <size_t SimdBitWidth, typename = void>
+struct intrin
+{
+    using vec_t = void*;
+
+    using mask_t = int;
+
+    static inline vec_t setzero() noexcept;
+
+    static inline vec_t set1_epi8(signed char w) noexcept;
+
+    static inline vec_t xor_vec(vec_t a, vec_t b) noexcept;
+
+    static inline vec_t and_vec(vec_t a, vec_t b) noexcept;
+
+    static inline vec_t or_vec(vec_t a, vec_t b) noexcept;
+
+    static inline vec_t load(const char* p) noexcept;
+
+    static inline bool equal(vec_t a, vec_t b) noexcept;
+
+    static inline mask_t less(vec_t a, vec_t b) noexcept;
+
+    static inline mask_t greater(vec_t a, vec_t b) noexcept;
+
+    static inline mask_t and_mask(mask_t a, mask_t b) noexcept;
+
+    static inline mask_t or_mask(mask_t a, mask_t b) noexcept;
+
+    static inline mask_t xor_mask(mask_t a, mask_t b) noexcept;
+
+    static inline auto to_underlying(mask_t a) noexcept;
+};
```

Comment:
You can ommit this things with the folllowing
```suggestion
template <size_t SimdBitWidth, typename T>
struct intrin{};
```

Reply:
This was purely for lsp autosuggestions.